### PR TITLE
refine: seperate parquet reader and arrow convert

### PR DESCRIPTION
**Pull request by ZENOTME**
This PR separates out the parquet reader from the arrow module. And make the arrow module a dir so that we can separate the `from_arow` and `to_arrow`.

### CODE REVIEW

**Comment by ZENOTME** [comment]
cc @liurenjie1024 @Xuanwo @Fokko

File: crates/iceberg/src/arrow/from.rs
Comment (liurenjie1024) [inline crates/iceberg/src/arrow/from.rs:1]: How about just name it as `schema.rs`, and we can put all schema related codes here?

File: crates/iceberg/src/reader.rs
Comment (liurenjie1024) [inline crates/iceberg/src/reader.rs:1]: Move this to `arrow` module?

**Review by liurenjie1024 — COMMENTED** [review]
Thanks @ZENOTME ! This refactoring look great to me, and I have some suggestions for naming part.

**Comment by liurenjie1024** [comment]
cc @viirya Would you also take a look?

**Review by viirya — COMMENTED** [review]
This looks good to me. I also suggest `schema.rs` instead of `from.rs`.

File: crates/iceberg/src/reader.rs
Comment (viirya) [reply crates/iceberg/src/reader.rs:1]: +1

**Review by liurenjie1024 — APPROVED** [review]
Thanks @ZENOTME , LGTM!

**Pull request merged**